### PR TITLE
PR: Reinstall spyder in editable mode in bootstrap if branch changed to/from master

### LINKED ISSUE #19728: Traceback from Editor/Debugger plugin when using same environment to launch different Spyder branches (master vs 5.x)
## Problem Description

When launching from master (previously having launched from 5.x on the same env) I got a traceback on the Anaconda prompt regarding the Debugger actions from the Editor. Maybe a validation is missing there or something is missing on the `bootstrap.py` script when changing branches? 

### What steps reproduce the problem?

1.  Launch Spyder from the master branch on an env that previously was used to launch the 5.x branch

### What is the expected output? What do you see instead?

No traceback

### Paste Traceback/Error Below (if applicable)

```python-traceback
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder\spyder\plugins\editor\plugin.py", line 1585, in setup_other_windows
    action = self.main.debugger.get_action(action_name)
AttributeError: 'NoneType' object has no attribute 'get_action'
```

## Versions

* Spyder version: 6.0.0.dev0 02aeb4d82 (conda)
* Python version: 3.10.5 64-bit
* Qt version: 5.15.3
* PyQt5 version: 5.15.4
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

When launching from bootstrap, detects whether the root branch has changed to/from master, in which case spyder is reinstalled in editable mode.

### Issue(s) Resolved

Fixes #19728

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @mrclary

### CODE REVIEW

File: bootstrap.py [hunk 1]
```diff
@@ -74,14 +72,35 @@
...
+    boot_branch_file = Path(get_conf_path("boot_branch.txt"))
+    if boot_branch_file.exists():
+        prev_branch = boot_branch_file.read_text()
+
+    revision, branch = vcs.get_git_revision(Path(__file__).parent)
+    boot_branch_file.write_text(branch)
+
+    logger.info("Previous branch: %s; current branch: %s", prev_branch, branch)
+
+    if "master" in (branch, prev_branch) and branch != prev_branch:
```
Comment (dalthviz) [inline bootstrap.py:92]: Maybe the Spyder editable reinstall should be forced even if the branch is not `master`. Thinking about it is possible that a branch is based on `master` or `5.x` without being exactly `master` or `5.x` right?

Comment (mrclary) [reply bootstrap.py]: I suppose what matters is whether the parent commit has changed to/from the master branch. Pushed up a change.

File: bootstrap.py [hunk 2]
```diff
@@ -74,14 +72,39 @@
...
+    ).communicate()
+    branch = "master" if res else "not master"
+    boot_branch_file.write_text(branch)
+
+    logger.info("Previous root branch: %s; current root branch: %s",
+                prev_branch, branch)
+
+    if branch != prev_branch:
+        logger.info("Detected root branch change to/from master. "
+                    "Will reinstall spyder in editable mode.")
```
Comment (ccordoba12) [inline bootstrap.py:98]:
```suggestion
                    "Will reinstall Spyder in editable mode.")
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @mrclary for your prompt response to address this issue.

File: bootstrap.py [hunk 3]
```diff
@@ -74,14 +72,39 @@
 
 # ---- Install sub repos
 
+from install_dev_repos import REPOS, install_repo
+from spyder import get_versions
+from spyder.config.base import get_conf_path
+from spyder.utils.programs import find_git, run_program
```
Comment (ccordoba12) [inline bootstrap.py:78]: Is it not possible to place these imports at the top of the file? If there's no technical reason that prevents it, I think they should be left there to follow our regular convention for imports.

**Review by ccordoba12 — COMMENTED** [review]
@mrclary, one more small thing I just noticed, then this should be ready on my side.

**Review by ccordoba12 — APPROVED** [review]
Thanks @mrclary!

**Review by dalthviz — APPROVED** [review]
Thanks @mrclary !

**Pull request merged**